expression/expression.go

```diff
@@ -773,12 +773,14 @@ func TableInfo2SchemaAndNames(ctx sessionctx.Context, dbName model.CIStr, tbl *m
 }
 
 // ColumnInfos2ColumnsAndNames converts the ColumnInfo to the *Column and NameSlice.
-func ColumnInfos2ColumnsAndNames(ctx sessionctx.Context, dbName, tblName model.CIStr, colInfos []*model.ColumnInfo, tblInfo *model.TableInfo) ([]*Column, types.NameSlice, error) {
-	columns := make([]*Column, 0, len(colInfos))
-	names := make([]*types.FieldName, 0, len(colInfos))
-	for i, col := range colInfos {
+func ColumnInfos2ColumnsAndNames(ctx sessionctx.Context, dbName, tblName model.CIStr, tblInfo *model.TableInfo, needWritableCols bool) ([]*Column, types.NameSlice, error) {
```

Comment:
- Because we want writableCols in [makeupDecodeColMap](https://github.com/pingcap/tidb/pull/19059/files#diff-279ebcd00bf6beb63507bc5b9a22457aR1194).

- Because `tblInfo` have not `WritableCols`.